app/services/resident_list.rb

```diff
@@ -0,0 +1,38 @@
+require 'pdf-reader'
+
+class InvalidFileFormatError < StandardError; end
+
+module ResidentList
+  class PDF
+    def initialize(file)
+      @file = file
+    end
+
+    def load_residents
+      # Guard clause to check if @file is a PDF
```

Comment:
You can remove these comments for the guard clauses.  You should have an empty line after guard clauses, which I think you generally do but just wanted to say so.